GeneralsMD/Code/GameEngine/Source/Common/System/BuildAssistant.cpp

```diff
@@ -871,22 +874,26 @@ LegalBuildCode BuildAssistant::isLocationClearOfObjects( const Coord3D *worldPos
 
 		// an immobile object will obstruct our building no matter what team it's on
 		if ( them->isKindOf( KINDOF_IMMOBILE ) )	{
+			Bool shrouded = them->getDrawable() && them->getDrawable()->getFullyObscuredByShroud();
```

Comment:
I think this condition will not work right when there is a Ghost Object. When a Ghost Object is shown, we want to see the Bib.

Reply:
If it's a ghost object then it is unknown whether the real object still exists and the area should thus require exploration.

Reply:
Original game shows bib on both Faction and Civilian building in shroud. I do not understand why we suddenly need to change that.

![shot_20250922_093921_1](https://github.com/user-attachments/assets/f180e684-0677-476e-a7f3-665c598f6109)

![shot_20250922_093925_2](https://github.com/user-attachments/assets/c6c96ce5-6545-4cd8-93cf-f06b5102c9f7)

In original game it does not show collision and Bib for Ghost Object, which reveals that this object has been sold (exploit). I think behaviour needs to be consistent and Ghost Object needs to show Bib as well as intact buildings.

![shot_20250922_094329_3](https://github.com/user-attachments/assets/483cf5cb-8cee-4d89-9533-f20537042317)

Reply:
There is no sudden need to change anything. It does not make sense to show bibs for shrouded objects as it reveals information about their existence.

Regardless, the bib is shown for the placement object with this change which is all that is necessary - similarly to attempting placement directly in fog - so there can be no confusion.